### PR TITLE
Update download script to use evaluated field + refactor of download script.

### PR DESCRIPTION
**Pull request by khanh111**

### CODE REVIEW

**Comment by piraka9011** [comment]
LGTM, awesome work! Will ping @abidlabs about this today.

File: download.py [hunk 1]
```diff
@@ -40,91 +48,112 @@
 
 WEBRTCVAD_SUPPORTED_SAMPLE_RATES_HZ = [8000, 16000, 32000, 48000]
 
-def download_csv_dataset():
-    print("Downloading CSV from", args.csv_url)
-    with requests.Session() as s:
-        download = s.get(args.csv_url)
+def does_cached_csv_dataset_exist(path_to_dataset_csv):
```
Comment (fdalvi) [inline download.py]: Should we move this + several of the following functions to a "dataset processing" module of its own? All of the methods that deal with checking/downloading/reading the dataset?

File: training/utils.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def get_one_hot_encoded_verse(surah_num, ayah_num):
     return decoder_input, decoder_target
 
 
-def shuffle_together(array1, array2, array3):
+def shuffle_together(array1, array2, array3, array4):
     """
     A helper method to randomly shuffle the order of three arrays while keeping their relative orders the same.
```
Comment (fdalvi) [inline training/utils.py]: update comment: order of *three -> four* arrays

We should consider writing a function that just takes arbitrary number of arrays using `*args` (incase some old code references the three array version)

File: download.py [hunk 3]
```diff
@@ -5,33 +5,41 @@
...
 parser = ArgumentParser(description='Tarteel Audio Downloader')
-parser.add_argument('--csv_url', type=str, default=TARTEEL_V1_CSV_URL)
-parser.add_argument('--local_csv_cache', type=str, default='.cache/local.csv')
-parser.add_argument('--local_download_dir', type=str, default='.audio')
-parser.add_argument('--vad_check', type=bool, default=True)
+parser.add_argument('--csv_url', type=str, default=TARTEEL_V1_LABELED_CSV_URL)
+parser.add_argument('--local_csv_filename', type=str, default='local.csv')
+parser.add_argument('--cache_dir', type=str, default='.cache')
+parser.add_argument('-u', '--use_cache', type=bool, default=False)
 parser.add_argument('-v', '--verbose', type=bool, default=False)
```
Comment (fdalvi) [inline download.py]: We should use `store_true` here for --verbose so it doesn't require an extra argument from the command line

Comment (fdalvi) [reply download.py]: Same for `--keep_downloaded_audio`

**Comment by fdalvi** [comment]
Actually I tried running the code using the following command
```bash
python download.py --cache_dir ../../data/juz_amma -s 78 -v True -k True
```
and it seems to completely ignore the `--cache_dir` (its empty once the script finishes), but the audio is saved in a `.tmp-RANDOM` directory where the script was run from - can you check why this is happening?

File: download.py [hunk 4]
```diff
@@ -40,91 +48,112 @@
...
-            os.remove(local_download_path)
+    surah_num, ayah_num, url, _, _, _, _, _, _, _, _ = entry
+
+    # Convert data to correct type from string.
+    surah_num = int(surah_num)
+    ayah_num = int(ayah_num)
+
+    # Ensure that the directories that the recording will be saved to exist.
+    downloaded_ayah_audio_dir = file_utils.prepare_ayah_directory(raw_audio_dir, surah_num, ayah_num)
+    ayah_audio_dir = file_utils.prepare_ayah_directory(raw_audio_dir, surah_num, ayah_num)
```
Comment (fdalvi) [inline download.py]: `downloaded_ayah_audio_dir` and `ayah_audio_dir` seem to be the same thing?

File: download.py (hunk 1 above)
Comment (khanh111) [reply download.py]: We have one and this function was an oversight, so I'll move it there.

File: download.py (hunk 3 above)
Comment (khanh111) [reply download.py]: Done. Should we keep downloaded audio by default? Can we listen to the raw files?

Comment (khanh111) [reply download.py]: If so, I'll change from `store_false` to `store_true` later.

File: download.py (hunk 4 above)
Comment (khanh111) [reply download.py]: Fixed.

File: training/utils.py (hunk 2 above)
Comment (khanh111) [reply training/utils.py]: Done.

**Pull request merged**